docs/developer-guide/typeConversions.md

```diff
@@ -0,0 +1,211 @@
+
+Conversion from strings to numeric values assume the string encodes a number otherwise it results in an invalid value.
+
+### Invalid Values
+
+Invalid values are returned if there is no value or the conversion is invalid for some reason, typically string to numeric value failures, in most cases an empty value equivalent is returned.
+
+- INT -> -9,223,372,036,854,775,808 ( `numeric_limits<std::int64_t>::min()`)
+- DOUBLE -> `-1e49`
+- COMPLEX -> {-1e49,0.0}
+- VECTOR -> {}
+- COMPLEX_VECTOR -> {}
+- STRING -> ""
+- NAMED_POINT -> {"", nan}
+- BOOL -> false
+- TIME -> Time::minVal() = `Time(numeric_limits<std::int64_t>::min())`
+
+## How types are used in HELICS
+
+### Interface specification
+
+For Value based interfaces types are used in a number of locations. For publications the most important part is specifying the type of the publication. This determines the type of data that gets transmitted. It can be a specific known type, a custom type string that is user defined or an `any` type.
+
+Inputs or subscriptions may also optionally specify a type as well. The use of this is for information to other federates, and for some type compatibility checking. It defaults to an "any" type so if nothing is specified it will match to any publication type.
+
+### Data publication and extraction
+
+As mentioned before when using any of the `publication` methods the extraction method call is not limited based on the type given in the interface specification. What the publication type does is define a conversion if necessary. The same is true on the output.
+
+SetVelueType -> PublicationType -> GetValueType
+
+Thus there are always two conversions occurring in the data translation pipeline. The first from the setValue to the type specified in the publication registration. The second is from the publish transmission type to the value requested in a respective getValue method.
+
+## Data Representation
+
+`int` and `double` are base level types in C++. `complex` is two doubles, `vector` and `complex_vector` are variable length arrays of doubles. `named_point` is `string` and double. String is a variable length sequence of 8 bit values. `char` is a string of length 1, `boolean` is a single char either `0` or `1`. Time is representation of internal HELICS time as a 64 bit integer (most commonly number of nanoseconds). The JSON type is a string compatible with JSON with two fields "type" and "value" The custom type is variable length sequence of bytes which HELICS will not attempt to convert it transmits a sequence of bytes it would be up to the user to define any conversions or endianness concerns.
```

Comment:
```suggestion
`int` and `double` are base level types in C++. `complex` is two doubles, `vector` and `complex_vector` are variable length arrays of doubles. `named_point` is `string` and double. String is a variable length sequence of 8 bit values. `char` is a string of length 1, `boolean` is a single char either `0` or `1`. Time is a representation of internal HELICS time as a 64 bit integer (most commonly number of nanoseconds). The JSON type is a string compatible with JSON with two fields "type" and "value". The custom type is a variable length sequence of bytes which HELICS will not attempt to convert, it transmits a sequence of bytes and it would be up to the user to define any conversions or endianness concerns.
```